app/Scripts/UpgradesHandler.gd

```diff
@@ -0,0 +1,19 @@
+extends Node
+
+export(Array, Resource) var presentUpgrades
+
+func AddAndApplyUpgrade(added_upgrade):
+	if added_upgrade != null:
+		presentUpgrades.append(added_upgrade)
+		if added_upgrade.has_method("Upgrade"):
+			added_upgrade.Upgrade()
+			PlayerStats.UpgradedValues()
```

Comment:
`PlayerStats.UpgradedValues()` would already have been called by line 9

Reply:
Yeah, I just did it just incase, I guess it's not needed